extension/settingsproviderextension/factory.go

```diff
@@ -0,0 +1,56 @@
+package settingsproviderextension
+
+import (
+	"context"
+	"fmt"
+	stsSettingsConfig "github.com/stackvista/sts-opentelemetry-collector/extension/settingsproviderextension/internal"
+	"github.com/stackvista/sts-opentelemetry-collector/extension/settingsproviderextension/provider/file"
+	stsSettingsSource "github.com/stackvista/sts-opentelemetry-collector/extension/settingsproviderextension/provider/kafka"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+)
+
+var (
+	Type = component.MustNewType("settings_provider")
+)
+
+func NewFactory() extension.Factory {
+	return extension.NewFactory(
+		Type,
+		createDefaultConfig,
+		createExtension,
+		// TODO: eventually this needs to become StabilityLevelStable
+		component.StabilityLevelDevelopment,
+	)
+}
+
+func createDefaultConfig() component.Config {
+	return &stsSettingsConfig.Config{
+		File: &stsSettingsConfig.FileSettingsProviderConfig{
+			Path: "./testdata/otel_mappings.yaml",
+		},
+	}
+}
+
+func createExtension(_ context.Context, set extension.CreateSettings, cfg component.Config) (extension.Extension, error) {
+	topoCfg := cfg.(*stsSettingsConfig.Config)
+	logger := set.Logger
+
+	if topoCfg.File != nil {
```

Comment:
Maybe first check that only 1 of the 2 is configured and fail with an error if that's the case.